### PR TITLE
Contine the JENKINS-27289 ticket: update the getResources as well

### PR DESCRIPTION
**Pull request by liorhson**
To @jglick request, Override the `getResources` method.
This PR continues the implementation from #1598 

Link to Jira: https://issues.jenkins-ci.org/browse/JENKINS-27289

### CODE REVIEW

File: core/src/main/java/hudson/util/MaskingClassLoader.java [hunk 1]
```diff
@@ -23,11 +23,9 @@
  */
 package hudson.util;
 
+import java.io.IOException;
 import java.net.URL;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
```
Comment (jglick) [inline core/src/main/java/hudson/util/MaskingClassLoader.java]: Generally prefer explicit to star imports, but I think we have no official style guide for the project…

File: core/src/main/java/hudson/util/MaskingClassLoader.java [hunk 2]
```diff
@@ -82,6 +80,16 @@ public synchronized URL getResource(String name) {
         return super.getResource(name);
     }
 
+    @Override
+    public Enumeration<URL> getResources(String name) throws IOException {
+        for (String mask : masksResources) {
+            if(name.startsWith(mask))
+                return null;
```
Comment (jglick) [inline core/src/main/java/hudson/util/MaskingClassLoader.java]: Is this even legal? Safer to return an empty enumeration.

File: core/src/main/java/hudson/util/MaskingClassLoader.java [hunk 3]
```diff
@@ -82,6 +80,16 @@ public synchronized URL getResource(String name) {
         return super.getResource(name);
     }
 
+    @Override
+    public Enumeration<URL> getResources(String name) throws IOException {
+        for (String mask : masksResources) {
+            if(name.startsWith(mask))
```
Comment (jglick) [inline core/src/main/java/hudson/util/MaskingClassLoader.java]: Could factor out a method `isMasked` from `getResource` to avoid duplication.

**Comment by liorhson** [comment]
You right about the empty enumeration.

BTW, in the last two LTS releases, #1598 wasn`t added, although it`s already merged, and is referenced Jira ticket (JENKINS-27289) labeled as  "lts-candidate".  Any news about that?

**Comment by daniel-beck** [comment]
@jglick Could you take a look at the updated PR?

**Comment by oleg-nenashev** [comment]
Looks good to me

**Comment by liorhson** [comment]
Great!
Can you guys merge this PR and his related one #1598 to the next Jenkins LTS 1.609.2 ?

**Comment by daniel-beck** [comment]
@liorhson Best to add that to the dev list thread you already posted to. Note that this is too recent to qualify through the regular process, so at least for this half of the fix, I wouldn't get my hopes up.

**Pull request merged**